cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -603,6 +614,7 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
                                 clusterCaConfig != null ? clusterCaConfig.getCertificateExpirationPolicy() : null);
                         clusterCa.createRenewOrReplace(
                                 reconciliation.namespace(), reconciliation.name(), caLabels.toMap(),
+                                clusterCaMetadata != null ? clusterCaMetadata.getLabels() : emptyMap(),
```

Comment:
The template has both labels and annotations. So I think you should pass both through and set them for the secret. Not just the labels.